--- a/toolchain/gcc/gcc-uclibc-4.x.mk	2010-09-01 00:07:06.000000000 +0200
+++ b/toolchain/gcc/gcc-uclibc-4.x.mk	2010-11-24 11:37:39.000000000 +0100
@@ -462,7 +462,7 @@
 GCC_BUILD_DIR4:=$(BUILD_DIR)/gcc-$(GCC_VERSION)-target
 
 $(GCC_BUILD_DIR4)/.prepared: $(STAMP_DIR)/gcc_libs_target_installed $(GCC_TARGET_PREREQ)
-	mkdir -p $(GCC_BUILD_DIR3)
+	mkdir -p $(GCC_BUILD_DIR4)
 	touch $@
 
 $(GCC_BUILD_DIR4)/.configured: $(GCC_BUILD_DIR4)/.prepared
