diff --git a/output/build/linux-3.2.6/drivers/rtc/rtc-mv.c b/output/build/linux-3.2.6/drivers/rtc/rtc-mv.c
--- a/output/build/linux-3.2.6/drivers/rtc/rtc-mv.c	2012-02-13 20:17:29.000000000 +0100
+++ b/output/build/linux-3.2.6/drivers/rtc/rtc-mv.c	2013-01-13 18:35:16.906433817 +0100
***************
*** 247,259 ****
  
  	/* make sure it is actually functional */
  	if (rtc_time == 0x01000000) {
! 		ssleep(1);
  		rtc_time = readl(pdata->ioaddr + RTC_TIME_REG_OFFS);
  		if (rtc_time == 0x01000000) {
  			dev_err(&pdev->dev, "internal RTC not ticking\n");
  			return -ENODEV;
  		}
! 	}
  
  	pdata->irq = platform_get_irq(pdev, 0);
  
--- 247,259 ----
  
  	/* make sure it is actually functional */
  	if (rtc_time == 0x01000000) {
! /* FOU		ssleep(1);
  		rtc_time = readl(pdata->ioaddr + RTC_TIME_REG_OFFS);
  		if (rtc_time == 0x01000000) {
  			dev_err(&pdev->dev, "internal RTC not ticking\n");
  			return -ENODEV;
  		}
! FOU */	}
  
  	pdata->irq = platform_get_irq(pdev, 0);
  
