--- a/package/microperl/microperl.mk
+++ b/package/microperl/microperl.mk
@@ -40,6 +40,7 @@ $(MICROPERL_DIR)/.host_configured_and_fixed: $(MICROPERL_DIR)/.host_configured
 	$(SED) 's/^.*<command-line>.*//g' $(MICROPERL_DIR)/Makefile
 	$(SED) 's/^.*<command-line>.*//g' $(MICROPERL_DIR)/x2p/Makefile
 	$(SED) 's/^.*<command-line>.*//g' $(MICROPERL_DIR)/makefile
+	$(SED) 's/^libs =/libs = -lm/g' $(MICROPERL_DIR)/makefile
 	$(SED) 's/^.*<command-line>.*//g' $(MICROPERL_DIR)/x2p/makefile
 	touch $@

