diff --git a/package/Config.in b/package/Config.in
index f308de7..5a7e349 100644
--- a/package/Config.in
+++ b/package/Config.in
@@ -90,6 +90,7 @@ endif
 source "package/sstrip/Config.in"
 if BR2_PACKAGE_BUSYBOX_SHOW_OTHERS
 source "package/tar/Config.in"
+source "package/mercurial/Config.in"
 endif
 endmenu
 
diff --git a/package/mercurial/Config.in b/package/mercurial/Config.in
new file mode 100644
index 0000000..cf466a4
--- /dev/null
+++ b/package/mercurial/Config.in
@@ -0,0 +1,10 @@
+config BR2_PACKAGE_MERCURIAL
+	bool "mercurial"
+	depends on BR2_PACKAGE_PYTHON && BR2_PACKAGE_PYTHON_BZIP2
+	help
+	  mercurial is a DVCS
+
+	  http://mercurial.selenic.com/
+
+comment "Mercurial requires a python interpreter with bzip2 enabled"
+    depends on !BR2_PACKAGE_PYTHON || !BR2_PACKAGE_PYTHON_BZIP2
diff --git a/package/mercurial/mercurial-001-no-host-python-bz2-check.patch b/package/mercurial/mercurial-001-no-host-python-bz2-check.patch
new file mode 100644
index 0000000..02fcc93
--- /dev/null
+++ b/package/mercurial/mercurial-001-no-host-python-bz2-check.patch
@@ -0,0 +1,26 @@
+--- mercurial-2.4.2.orig/setup.py	2013-01-02 07:07:38.000000000 +0100
++++ mercurial-2.4.2/setup.py	2013-01-28 21:38:30.288042140 +0100
+@@ -36,23 +36,6 @@
+     raise SystemExit(
+         "Couldn't import standard zlib (incomplete Python install).")
+ 
+-# The base IronPython distribution (as of 2.7.1) doesn't support bz2
+-isironpython = False
+-try:
+-    isironpython = (platform.python_implementation()
+-                    .lower().find("ironpython") != -1)
+-except AttributeError:
+-    pass
+-
+-if isironpython:
+-    sys.stderr.write("warning: IronPython detected (no bz2 support)\n")
+-else:
+-    try:
+-        import bz2
+-    except ImportError:
+-        raise SystemExit(
+-            "Couldn't import standard bz2 (incomplete Python install).")
+-
+ import os, subprocess, time
+ import shutil
+ import tempfile
diff --git a/package/mercurial/mercurial.mk b/package/mercurial/mercurial.mk
new file mode 100644
index 0000000..a46d8d2
--- /dev/null
+++ b/package/mercurial/mercurial.mk
@@ -0,0 +1,27 @@
+#############################################################
+#
+# mercurial
+#
+#############################################################
+
+MERCURIAL_VERSION = 2.4.2
+MERCURIAL_SOURCE  = mercurial-$(MERCURIAL_VERSION).tar.gz
+MERCURIAL_SITE    = http://mercurial.selenic.com/release
+
+MERCURIAL_DEPENDENCIES = python
+
+define MERCURIAL_BUILD_CMDS
+	(cd $(@D); \
+		CC="$(TARGET_CC)"		\
+		CFLAGS="$(TARGET_CFLAGS) -I$(STAGING_DIR)/usr/include/python$(PYTHON_VERSION_MAJOR)" 	\
+		LDSHARED="$(TARGET_CC) -shared" \
+		LDFLAGS="$(TARGET_LDFLAGS)" 	\
+	$(HOST_DIR)/usr/bin/python setup.py build)
+endef
+
+define MERCURIAL_INSTALL_TARGET_CMDS
+	(cd $(@D); $(HOST_DIR)/usr/bin/python setup.py install --prefix=$(TARGET_DIR)/usr)
+    (cd $(@D); sed -i -e 's/^#!.*\/usr\/bin\/python/#!\/usr\/bin\/python/' $(TARGET_DIR)/usr/bin/hg)
+endef
+
+$(eval $(generic-package))
