diff --git a/Makefile b/Makefile
index f4f7514..12930f2 100644
--- a/Makefile
+++ b/Makefile
@@ -442,10 +442,10 @@ ifneq ($(BR2_HAVE_DOCUMENTATION),y)
 	-rmdir $(TARGET_DIR)/usr/share 2>/dev/null
 endif
 ifeq ($(BR2_PACKAGE_PYTHON_PY_ONLY),y)
-	find $(TARGET_DIR)/usr/lib/ -name '*.pyc' -print0 | xargs -0 rm -f
+	find $(TARGET_DIR)/usr/lib/python$(PYTHON_VERSION_MAJOR) -name '*.pyc' -print0 | xargs -0 rm -f
 endif
 ifeq ($(BR2_PACKAGE_PYTHON_PYC_ONLY),y)
-	find $(TARGET_DIR)/usr/lib/ -name '*.py' -print0 | xargs -0 rm -f
+	find $(TARGET_DIR)/usr/lib/python$(PYTHON_VERSION_MAJOR) -name '*.py' -print0 | xargs -0 rm -f
 endif
 	$(STRIP_FIND_CMD) | xargs $(STRIPCMD) 2>/dev/null || true
 	find $(TARGET_DIR)/lib/modules -type f -name '*.ko' | \
