--- a/package/dropbear/dropbear.mk	2010-09-01 00:07:06.000000000 +0200
+++ b/package/dropbear/dropbear.mk	2010-12-15 22:38:55.000000000 +0100
@@ -21,6 +21,7 @@
 
 $(DROPBEAR_HOOK_POST_EXTRACT):
 	$(SED) 's,^#define XAUTH_COMMAND.*/xauth,#define XAUTH_COMMAND "/usr/bin/xauth,g' $(DROPBEAR_DIR)/options.h
+	$(SED) 's,^#define DO_HOST_LOOKUP.*,/* #define DO_HOST_LOOKUP */,' $(DROPBEAR_DIR)/options.h
 	touch $@
 
 $(DROPBEAR_TARGET_INSTALL_TARGET):
