diff --git a/package/Config.in b/package/Config.in
index 132b9c3..dfc36f5 100644
--- a/package/Config.in
+++ b/package/Config.in
@@ -279,6 +279,9 @@ source "package/python-setuptools/Config.in"
 endmenu
 endif
 source "package/python3/Config.in"
+if BR2_PACKAGE_PYTHON3
+source "package/python3-serial/Config.in"
+endif
 source "package/ruby/Config.in"
 source "package/tcl/Config.in"
 endmenu
diff --git a/package/python3-serial/Config.in b/package/python3-serial/Config.in
new file mode 100644
index 0000000..7b1fcf8
--- /dev/null
+++ b/package/python3-serial/Config.in
@@ -0,0 +1,8 @@
+config BR2_PACKAGE_PYTHON3_SERIAL
+	bool "python3-serial"
+	depends on BR2_PACKAGE_PYTHON3
+	help
+	  python-serial is a Python library to access serial ports.
+
+	  http://pyserial.sf.net
+
diff --git a/package/python3-serial/python3-serial.mk b/package/python3-serial/python3-serial.mk
new file mode 100644
index 0000000..7f6d0d5
--- /dev/null
+++ b/package/python3-serial/python3-serial.mk
@@ -0,0 +1,21 @@
+#############################################################
+#
+# python3-serial
+#
+#############################################################
+
+PYTHON3_SERIAL_VERSION = 2.6
+PYTHON3_SERIAL_SOURCE  = pyserial-$(PYTHON3_SERIAL_VERSION).tar.gz
+PYTHON3_SERIAL_SITE    = http://pypi.python.org/packages/source/p/pyserial/
+
+PYTHON3_SERIAL_DEPENDENCIES = python3
+
+define PYTHON3_SERIAL_BUILD_CMDS
+	(cd $(@D); $(HOST_DIR)/usr/bin/python3 setup.py build)
+endef
+
+define PYTHON3_SERIAL_INSTALL_TARGET_CMDS
+	(cd $(@D); $(HOST_DIR)/usr/bin/python3 setup.py install --prefix=$(TARGET_DIR)/usr)
+endef
+
+$(eval $(generic-package))
